gremlin-go/driver/result.go

```diff
@@ -21,18 +21,149 @@ package gremlingo
 
 // AN-968 Finish Result implementation.
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+)
 
 // Result Struct to abstract the Result and provide functions to use it.
 type Result struct {
 	result interface{}
 }
 
-// AsString returns the current Result formatted as a string.
+func newResult(result interface{}) *Result {
+	return &Result{result}
+}
+
+// AsString returns the current result item formatted as a string.
 func (r *Result) AsString() string {
 	return fmt.Sprintf("%v", r.result)
 }
 
-func newResult(result interface{}) *Result {
-	return &Result{result}
+// ToString returns the string representation of the Result struct in Go-syntax format
+func (r *Result) ToString() string {
+	return fmt.Sprintf("result{object=%v class=%T}", r.result, r.result)
+}
+
+// GetString gets the string representation of the result, equivalent to AsString()
```

Comment:
With this in mind, should we drop AsString() and have the functions invoke GetString()?

Reply:
I think so. I don't see a reason for this extra function at this point.